text/0298-deprecate-component-lifecycle-events.md

```diff
@@ -0,0 +1,103 @@
+- Start Date: 2018-01-19
+- RFC PR: https://github.com/emberjs/rfcs/pull/298
+- Ember Issue: (leave this empty)
+
+# Deprecate Component Lifecycle Events
+
+## Summary
+
+The goal of this RFC is to remove component lifecycle events, such as `Ember.on('didRender'`,
+in favor of enforcing usage of the lifecycle methods themselves i.e. `didRender(){}`
+
+## Motivation
+
+When using events, timing is not guaranteed, so it is possible to have unintended
+consequences and have things execute in an unexpected order. Using the methods directly
+does guarantee timing and solidifies the path forward on one recommended way of using
+lifecycle hooks.
+
+## Detailed design
+
+Ember will start logging deprecation messages that state which event you used and which
+method it should be converted to.
+
+The exact deprecation message will be decided later, but something along the lines of:
+
+```
+Using `Ember.on` event listeners for component lifecycle hooks is deprecated.
+Please move your code from `myHandler: Ember.on('didRender'` into the `didRender(){}` method.
+```
+
+Ember does not appear to rely on these events internally, however there are at least a couple of mentions of 
+these events in the comments, referring specifically to `on('didInsertElement')` 
+[here](https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/run_loop.js#L140) 
```

Comment:
should change these to permalinks at a tag, so they won't be dead in the future!

Reply:
Good call! Updating now.